### PR TITLE
[TVMScript] IRBuilder methods for `IRModule`

### PR DESCRIPTION
**Pull request by cyx-6**
This PR introduces IRBuilder methods for `IRModule`.

Co-authored-by: yongwww <yongcale@gmail.com>

### CODE REVIEW

File: include/tvm/script/ir_builder/ir/frame.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include <tvm/node/node.h>
+#include <tvm/script/ir_builder/base.h>
+
+#include <vector>
+
+namespace tvm {
+namespace script {
+namespace ir_builder {
+
+class IRModuleFrameNode : public IRBuilderFrameNode {
```
Comment (junrushao) [inline include/tvm/script/ir_builder/ir/frame.h:33]: Please add some docs to this class

File: include/tvm/script/ir_builder/ir/ir.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#include <tvm/node/node.h>
+#include <tvm/script/ir_builder/ir/frame.h>
+
+#include <vector>
+
+namespace tvm {
+namespace script {
+namespace ir_builder {
+
+TVM_DLL IRModuleFrame IRModule();
```
Comment (junrushao) [inline include/tvm/script/ir_builder/ir/ir.h:33]: ditto. Let's add some docs

File: include/tvm/script/ir_builder/ir/frame.h (hunk 1 above)
Comment (cyx-6) [reply include/tvm/script/ir_builder/ir/frame.h:38]: added

File: include/tvm/script/ir_builder/ir/ir.h (hunk 2 above)
Comment (cyx-6) [reply include/tvm/script/ir_builder/ir/ir.h:37]: added

**Review by yongwww — APPROVED** [review]
LGTM!

**Review by yongwww — APPROVED** [review]

File: tests/python/unittest/test_tvmscript_ir_builder_irmodule.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    ir_module_actual = ib.get()
+
+    # the expected prim_func
+    ir_module_expected = ir.IRModule(None, None)
+
+    assert_structural_equal(ir_module_actual, ir_module_expected, map_free_vars=True)
+
+
+if __name__ == "__main__":
+    test_ir_builder_irmodule()
```
Comment (vinx13) [inline tests/python/unittest/test_tvmscript_ir_builder_irmodule.py:40]: nit: it's preferred to use `tvm.testing.main()` here

Comment (cyx-6) [reply tests/python/unittest/test_tvmscript_ir_builder_irmodule.py]: Thanks and updated!

**Review by vinx13 — APPROVED** [review]

**Pull request merged**